### PR TITLE
[1.9] - Fix Deployment with Recreate strategy not to wait on Pods in terminal phase

### PR DESCRIPTION
**Pull request by tnozicka**
Backport of https://github.com/kubernetes/kubernetes/pull/60301

```release-note
Fixes a case when Deployment with recreate strategy could get stuck on old failed Pod.
```

/assign @janetkuo @kow3ns

### CODE REVIEW

File: pkg/controller/deployment/recreate_test.go [hunk 1]
```diff
@@ -90,34 +90,138 @@ func TestOldPodsRunning(t *testing.T) {
...
+			},
+			hasOldPodsRunning: true,
 		},
 	}
 
 	for _, test := range tests {
-		if expected, got := test.expected, oldPodsRunning(test.newRS, test.oldRSs, test.podMap); expected != got {
-			t.Errorf("%s: expected %t, got %t", test.name, expected, got)
-		}
+		t.Run(test.name, func(t *testing.T) {
```
Comment (janetkuo) [inline pkg/controller/deployment/recreate_test.go]: Remove the use of `t.Run` (only keep relevant changes)

Comment (tnozicka) [reply pkg/controller/deployment/recreate_test.go]: you are right; I didn't pay enough attention while resolving that conflict
fixed

**Review by janetkuo — APPROVED** [review]

**Comment by janetkuo** [comment]
@tnozicka not sure if the verify failure is relevant. Would you take a look?

**Comment by tnozicka** [comment]
@janetkuo pull-kubernetes-verify is failing in all branches, someone fixed it in master lately by updating some bazel related component, but not here

(It's failing even here where 1 bash line is changed https://github.com/kubernetes/kubernetes/pull/60494)

**Comment by tnozicka** [comment]
I was told that https://github.com/kubernetes/kubernetes/pull/60522 should fix it

**Comment by mbohlool** [comment]
/hold

why this is not a an automatic cherrypick?

**Comment by janetkuo** [comment]
> why this is not a an automatic cherrypick?

@tnozicka did you use the cherrypick script to create this PR? https://github.com/kubernetes/community/blob/master/contributors/devel/cherry-picks.md#how-do-cherrypick-candidates-make-it-to-the-release-branch

**Comment by tnozicka** [comment]
I haven't used that script, nor I see it to be required. 

That mention in the article looks like guidance only and some people might not feel comfortable running a bash script messing with their git. I am not.

I used `git cherry-pick` as many people do looking at http://cherrypick.k8s.io/#/queue

@mbohlool why would you hold a PR on that? Is there something wrong with it?

**Comment by mbohlool** [comment]
The branch is frozen. holding this for 1.9.5.

/hold

**Comment by thockin** [comment]
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/60493)

**Pull request merged**